pkg/dhcpd/dhcpd.go

```diff
@@ -0,0 +1,110 @@
+package dhcpd
+
+import (
+	"fmt"
+	dhcp "github.com/krolaw/dhcp4"
+	"github.com/krolaw/dhcp4/conn"
+	"github.com/samber/lo"
+	"github.com/vinted/rest-dhcpd/pkg/configdb"
+	"github.com/vinted/rest-dhcpd/pkg/rest"
+	"log"
+	"net"
+	"strconv"
+	"time"
+)
+
+type DHCPHandler struct {
+	ip            net.IP
+	leaseDuration time.Duration
+	options       dhcp.Options
+}
+
+var data_type = map[int]string{
+	// List of Options that stores IP address represented in 4 bytes of data (without dots).
+	1:  "IP",
+	2:  "IP",
+	3:  "IP",
+	4:  "IP",
+	5:  "IP",
+	6:  "IP",
+	7:  "IP",
+	8:  "IP",
+	9:  "IP",
+	10: "IP",
+	11: "IP",
+}
+
+func StartServer() {
+	lease, _ := time.ParseDuration(strconv.Itoa(configdb.Config.LeaseDuration) + "s")
+	handler := &DHCPHandler{
+		ip:            net.ParseIP(configdb.Config.IP),
+		leaseDuration: lease,
+		options:       BuildOptions(configdb.Config.Options),
+	}
+	cn, err := conn.NewUDP4BoundListener(configdb.Config.ListenInterface, ":67")
+	if err != nil {
+		log.Fatal("NewUDP4BoundListener")
```

Comment:
We may want a better error message here. Would be nice to know if port is already in use or something else

Reply:
@lukassup fixed